kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/HttpClientUtils.java

```
@@ -18,167 +18,184 @@
+        } catch (GeneralSecurityException e) {
+          throw new AssertionError(); // The system has no TLS. Just give up.
+        }
+      } else {
+        SSLContext context = SSLContext.getInstance("TLSv1.2");
+        context.init(keyManagers, trustManagers, null);
+        httpClientBuilder.sslSocketFactory(context.getSocketFactory(), (X509TrustManager) trustManagers[0]);
+      }
+
+      httpClientBuilder.addInterceptor(new Interceptor() {
+        @Override
+        public Response intercept(Chain chain) throws IOException {
+          final int maxRetries = config.getMaxRetries();
+          final int retryDelay = config.getRetryDelayInSeconds();
+          Response response = null;
+          int retries = 0;
+          do {
+            try {
               if (isNotNullOrEmpty(config.getUsername()) && isNotNullOrEmpty(config.getPassword())) {
+                LOGGER.info("Using username and password Authorization");
                 Request authReq = chain.request().newBuilder().addHeader("Authorization", Credentials.basic(config.getUsername(), config.getPassword())).build();
-                return chain.proceed(authReq);
+                response = chain.proceed(authReq);
               } else if (isNotNullOrEmpty(config.getOauthToken())) {
+                LOGGER.info("Using OAuth Authorization");
                 Request authReq = chain.request().newBuilder().addHeader("Authorization", "Bearer " + config.getOauthToken()).build();
-                return chain.proceed(authReq);
+                response = chain.proceed(authReq);
+              }
+            } catch (ConnectException e) {
+              LOGGER.warn("Request failed retry {} ", retries);
+              try {
+                SECONDS.sleep(retryDelay);
+              } catch (InterruptedException e1) {
+                //dont worry
               }
-              return chain.proceed(request);
+              retries++;
+            } catch (IOException e) {
```

In the same spirit I wouldn't even catch that exception and would let it propagate.